src/predicates/intersects.jl

```diff
@@ -33,6 +33,20 @@ intersects(r::Ray, t::Triangle) = !isnothing(r ∩ t)
 
 intersects(t::Triangle, r::Ray) = intersects(r, t)
 
+function intersects(r::Ray, s::Sphere)
+  s₀ = center(s) - r(0)
+  h = hypot(s₀...)
+  radius(s) > h && return true
+  r₀ = r(1) - r(0)
+  return abs(∠(s₀, r₀)) < asin(radius(s) / h)
```

Comment:
```suggestion
  abs(∠(s₀, r₀)) < asin(radius(s) / h)
```
Is it possible to rewrite this in terms of `atan2`? Not a requirement, but maybe can improve the stability on the edge cases.

Reply:
I don't really think so. I mean, geometrically we could, but it would add more complexity to the calculation, and I actually think this simple case nicely encapsulates the question of "what's maximum angular size of a sphere, when measured from the base of the ray?" Perhaps a doc-string explaining the geometry, and perhaps even a link to a diagram (similar to the one I attached) would go a long way to explaining the algorithm in a concise way. 

The edge case of "what if the base of the ray is inside the sphere?" is handled on the early return on line 39, and the only other edge cases I can imagine are if `hypot(s₀...)` or `radius(s)` are somehow negative. But in my estimation, I think we'd want to treat those cases the same as the positive value, which `asin` will do, but `atan2` will not. 

Naturally, my imagination is not boundless, and the possibility of bugs-from-edge-cases always seems to be 😉. I am happy to further arguments in favour of the atan2 case, of course.

Reply:
No problem at all. We can stick to asin as suggested. I just wanted to double check with you first.